src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSService.java

```diff
@@ -329,7 +331,7 @@ TaskDefinition registerTemplate(final String cloudName, final ECSTaskTemplate te
                 request.withExecutionRoleArn(template.getExecutionRole());
             }
 
-            if (!StringUtils.isEmpty(template.getTaskrole())) {
+            if (!StringUtils.isEmpty(template.getTaskrole()) && !template.getUseSessionTags()) {
```

Comment:
If we're using session tags, then don't set the task role ARN as we will pass creds in via environment variables